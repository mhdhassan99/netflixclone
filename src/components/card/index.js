import React, { useState, useContext, createContext } from 'react';

export const FeatureContext = createContext();

export default function Card({ children, ...restProps }) {
    const [showFeature, setShowFeature] = useState(false);
    const [itemFeature, setItemFeature] = useState({});

    return (
        <FeatureContext.Provider
            value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
        >
            <Container {...restProps}>{children}</Container>
        </FeatureContext.Provider>
    )
};

Card.Group = function CardGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}