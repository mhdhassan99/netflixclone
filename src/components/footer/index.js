import React from 'react';

const Footer = ({ children, ...restProps }) => {
    return (
        <Container {...restProps}>
            {children}
        </Container>
    );
}
export default Footer;

Footer.Row = function FooterRow({ children, ...restProps }) {
    return <Row {...restProps}> {children} </Row>
} 

Footer.Column = function FooterColumn({ children, ...restProps }) {
    return <Column {...restProps}> {children} </Column>
} 

Footer.Link = function FooterLink({ children, ...restProps }) {
    return <Link {...restProps}> {children} </Link>
} 

Footer.Title = function FooterTitle({ children, ...restProps }) {
    return <Title {...restProps}> {children} </Title>
} 
