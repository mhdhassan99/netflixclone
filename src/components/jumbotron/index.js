import React from 'react';
import { Inner } from './styles/jumbotron';

const Jumbotron = ({children, direction = 'row', ...restProps }) => {
    return(
        <Inner direction={direction}>
            <p>I am Jumbo</p>
        </Inner>
    )
}

export default Jumbotron;
