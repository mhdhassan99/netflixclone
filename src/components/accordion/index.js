import React, {useState} from 'react';
import { Container, Inner } from '../jumbotron/styles/jumbotron';

const Accordion = ({ children, ...restProps }) => {
    return (
        <Container {...restProps}>
            <Inner>{children}</Inner>
        </Container>
    );
}
export default Accordion;

Accordion.Title = function AccordionTitle({ children, ...restProps}) {
    return <Title { ...restProps }>{children}</Title>
}; 

Accordion.Frame = function AccordionFrame({ children, ...restProps}) {
    return <Frame { ...restProps }>{children}</Frame>
}; 

Accordion.Item = function AccordionItem({ children, ...restProps}) {
    const [toggleShow, setToggleShow] = useState(false)

    return <Item { ...restProps }>{children}</Item>
}; 

Accordion.Header = function AccordionHeader({ children, ...restProps}) {
    return <Header { ...restProps }>{children}</Header>
}; 