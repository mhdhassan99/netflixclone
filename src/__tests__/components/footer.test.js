import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from '../../components';

describe ('<Footer />', () => {
    it('renders the <Footer /> with populated data', () => {
        const { container, getByText } = render (
            <Footer>
                <Footer.Title>Questions? Contact us.</Footer.Title>
                <Footer.Break />
                <Footer.Row>
                    <Footer.Column>
                        <Footer.Link herf="#">FAQ</Footer.Link>
                        <Footer.Link herf="#">Investor Relations</Footer.Link>
                        <Footer.Link herf="#">Ways to Watch</Footer.Link>
                        <Footer.Link herf="#">Corporate Information</Footer.Link>
                        <Footer.Link herf="#">Netflix Originals</Footer.Link>
                    </Footer.Column>

                    <Footer.Column>
                        <Footer.Link herf="#">Help center</Footer.Link>
                        <Footer.Link herf="#">Terms of use</Footer.Link>
                        <Footer.Link herf="#">Jobs</Footer.Link>
                        <Footer.Link herf="#">Contact us</Footer.Link>
                    </Footer.Column>

                    <Footer.Column>
                        <Footer.Link herf="#">Account</Footer.Link>
                        <Footer.Link herf="#">Redeem Gift Cards</Footer.Link>
                        <Footer.Link herf="#">Privacy</Footer.Link>
                        <Footer.Link herf="#">Speed Test</Footer.Link>
                    </Footer.Column>

                    <Footer.Column>
                        <Footer.Link herf="#">Media Center</Footer.Link>
                        <Footer.Link herf="#">Buy Gift Cards</Footer.Link>
                        <Footer.Link herf="#">Cookies Preferences</Footer.Link>
                        <Footer.Link herf="#">Legal Notice</Footer.Link>
                    </Footer.Column>
                </Footer.Row>
                <Footer.Break/>
                <Footer.Text>Netflix United States</Footer.Text>
            </Footer>
        )
        expect(getByText('Questions? Contact us.')).toBeTruthy();
        expect(getByText('FAQ')).toBeTruthy();
        expect(getByText('Investor Relations')).toBeTruthy();
        expect(getByText('Ways to Watch')).toBeTruthy();
        expect(getByText('Corporate Information')).toBeTruthy();
        expect(getByText('Netflix Originals')).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });
});