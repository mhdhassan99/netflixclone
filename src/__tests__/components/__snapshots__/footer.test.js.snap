// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-qu6i6-0 cyHgzK"
>
  <p
    class="footer__Title-qu6i6-4 ctavuJ"
  >
     
    Questions? Contact us.
     
  </p>
  <p
    class="footer__Break-qu6i6-6 cSuQKC"
  >
     
     
  </p>
  <div
    class="footer__Row-qu6i6-2 eQpqDY"
  >
     
    <div
      class="footer__Column-qu6i6-1 jAYoWp"
    >
       
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        FAQ
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Investor Relations
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Ways to Watch
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Corporate Information
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Netflix Originals
         
      </a>
       
    </div>
    <div
      class="footer__Column-qu6i6-1 jAYoWp"
    >
       
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Help center
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Terms of use
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Jobs
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Contact us
         
      </a>
       
    </div>
    <div
      class="footer__Column-qu6i6-1 jAYoWp"
    >
       
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Account
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Redeem Gift Cards
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Privacy
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Speed Test
         
      </a>
       
    </div>
    <div
      class="footer__Column-qu6i6-1 jAYoWp"
    >
       
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Media Center
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Buy Gift Cards
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Cookies Preferences
         
      </a>
      <a
        class="footer__Link-qu6i6-3 iRyBkK"
      >
         
        Legal Notice
         
      </a>
       
    </div>
     
  </div>
  <p
    class="footer__Break-qu6i6-6 cSuQKC"
  >
     
     
  </p>
  <p
    class="footer__Text-qu6i6-5 bAqYDB"
  >
     
    Netflix United States
     
  </p>
</div>
`;
